### PR TITLE
listens for pdf and uploads to wiki

### PR DESCRIPTION
**Pull request by hydrowoxy**
Listens for pdfs in the #notes-and-stuff channel and automatically puts them on the wiki under the Meeting Minutes folder/page

### CODE REVIEW

File: events/pdf-listener.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const { Events } = require('discord.js');
+const FormData = require('form-data');
+const { wikiApiKey } = require('../config.json');
+const axios = require('axios');
+
+module.exports = {
+    name: Events.MessageCreate,
+    async execute(message) {
+        if (message.channel.id === '1298058637121556612' && message.attachments.size > 0) { // Channel to listen in
```
Comment (gr812b) [inline events/pdf-listener.js:9]: Setting a const to `'1298058637121556612'` and giving it a descriptive name would make this easier to read. Could leave it as a global variable underneath imports

File: events/pdf-listener.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            await Promise.all(message.attachments.map(async (attachment) => {
+                if (attachment.contentType !== 'application/pdf') return;
+                console.log('PDF detected:', attachment.url);
+
+                try {
+                    const { data } = await axios.get(attachment.url, { responseType: 'arraybuffer' });
+                    const formData = new FormData();
+
+                    formData.append('mediaUpload', JSON.stringify({ folderId: 17 }), { contentType: 'application/json' }); // 17 is meeting-minutes folder
```
Comment (gr812b) [inline events/pdf-listener.js:18]: Same as above, nitpicky but instead of comments, make a const

File: events/pdf-listener.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                            }`
+                        }); // New page to be added with graphQL
+
+                        const pageResponse = await axios.post('http://wiki.mcmasterbaja.ca/graphql', pageMutation, {
+                            headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${wikiApiKey}` },
+                        });
+                        console.log(pageResponse.status === 200 ? 'Page created successfully.' : 'Failed to create page:', pageResponse.data);
+                    }
+                } catch (error) {
+                    console.error('Error processing PDF:', error.message);
```
Comment (gr812b) [inline events/pdf-listener.js:53]: Might be a nice idea to send a message that the document could not be uploaded inside the notes and stuff channel. Could even also add an ephemeral "Sucessfully made new page etc" message on success too as a reply

File: events/pdf-listener.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                                  isPublished: true,
+                                  isPrivate: false,
+                                  locale: "en",
+                                  path: "/meeting-minutes/${attachment.name.replace('.pdf', '')}",
+                                  tags: []
+                                ) {
+                                  responseResult { message }
+                                }
+                              }
+                            }`
```
Comment (gr812b) [inline events/pdf-listener.js:44]: Nit: Prob not a bad idea to define this above for readability

**Review by gr812b — CHANGES_REQUESTED** [review]
A few nitpicks that could be addressed or ignored, and then I think adding the feedback to the user would be very beneficial and so requesting that pls 😍

**Review by gr812b — APPROVED** [review]
Looks great!

**Comment by hydrowoxy** [comment]
yippee!!!

**Pull request merged**